third_party/terraform/utils/iam.go

```diff
@@ -123,6 +123,25 @@ func iamPolicyReadModifyWrite(updater ResourceIamUpdater, modify iamPolicyModify
 	return nil
 }
 
+// Takes a single binding and will either overwrite the same role in a list or append it to the end
+func overwriteBinding(bindings []*cloudresourcemanager.Binding, overwrite *cloudresourcemanager.Binding) []*cloudresourcemanager.Binding {
```

Comment:
Like above - this reads almost identically to `Update` in IAM Binding. A PR introducing this should include a reason why we couldn't use the exact same code for both.